File: Sliding Window/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> mpp(256, -1);
        int res = 0, left = 0, right = 0, len = 0;
        while (right < (int)s.size()) {
            if (mpp[s[right]] != -1) {
                left = max(left,mpp[s[right]]+1);
            }
            mpp[s[right]] = right;
            len = max(len, right - left + 1);
            right++;
        }
        return len;
    }
};

File: Sliding Window/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(s: str) -> int:
        setChar = set()
        n = len(s)
        l = 0
        res = 0
        for r in range(n):
            while s[r] in setChar:
                setChar.remove(s[l])
                l+=1
            setChar.add(s[r])
            res = max(res,r-l+1)
        return res
